nixos/modules/programs/sway-beta.nix

```diff
@@ -0,0 +1,33 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.sway-beta;
+  swayPackage = cfg.package;
+in {
+  options.programs.sway-beta = {
+    enable = mkEnableOption ''
+      The i3-compatible Wayland compositor. (temporary beta package)
+    '';
+
+    package = mkOption {
+      type = types.package;
+      default = pkgs.sway-beta;
+      defaultText = "pkgs.sway-beta";
+      description = ''
+        The package that will be used for sway.
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    environment.systemPackages = [ swayPackage ];
```

Comment:
IIRC `xwayland` is required in `environment.systemPackages` (or PATH) for XWayland support (didn't test this but yesterday I got an error message without it). Maybe we should keep `extraPackages` (we could also suggest termite, rofi, ...) via the examples)?